common/display/displayqueue.cpp

```diff
@@ -155,26 +144,28 @@ bool DisplayQueue::ApplyPendingModeset(drmModeAtomicReqPtr property_set) {
 }
 
 bool DisplayQueue::SetPowerMode(uint32_t power_mode) {
-  ScopedSpinLock lock(spin_lock_);
-
-  if ((power_mode == kOff) || (power_mode == kDoze)) {
-    dpms_mode_ = DRM_MODE_DPMS_OFF;
-    Flush();
-    HWCThread::ConditionalSuspend();
-    drmModeConnectorSetProperty(gpu_fd_, connector_, dpms_prop_, dpms_mode_);
-    return true;
-  }
-
-  if (power_mode == kDozeSuspend) {
-    Flush();
-    HWCThread::ConditionalSuspend();
-    return true;
-  }
-
-  if (power_mode == kOn) {
-    dpms_mode_ = DRM_MODE_DPMS_ON;
-    needs_modeset_ = true;
-    drmModeConnectorSetProperty(gpu_fd_, connector_, dpms_prop_, dpms_mode_);
+  switch (power_mode) {
+    case kOff:
```

Comment:
Atleast didnt show anything weird in this case and we could go immediately to power off in this case, since we release all resources. When are powered on again, we dont have access to last buffer and application has to provide us with the content again, thats why got rid of flush here.